Model/ResourceModel/Magento/Product/CollectionBuilder.php

```diff
@@ -173,14 +172,14 @@ public function setCurrentPage($currentPage)
     }
 
     /**
-     * Sets the default visibility. Calls self::withOnlyVisibleInSites()
-     * and self::withOnlyActiveAndVisible()
+     * Sets the default visibility and set active products filter based on configuration
      *
+     * @param Store $store
      * @return CollectionBuilder
      */
-    public function withDefaultVisibility()
+    public function withDefaultVisibility(Store $store)
     {
-        return $this->withOnlyVisibleInSites()->withOnlyActiveAndVisible();
+        return $this->withConfiguredProductStatus($store);
```

Comment:
We should not index invisible products. This will end up putting all skus in the index queue and slow down the indexing process.

Reply:
Understood 👍

Reply:
Problem: collection excludes disabled products when visibility filter is added.